src/pattern.ts

```diff
@@ -0,0 +1,36 @@
+import { string, oneOf, object } from "@mojotech/json-type-validation"
+
+export interface RegexPattern {
+  regex: RegExp
+}
+export type LiteralPattern = string
+export type Pattern = RegexPattern | LiteralPattern
+
+export function matchesPattern (pattern: Pattern, input: string) {
+  if (typeof pattern === 'string') {
+    return pattern === input
+  } else if ('regex' in pattern) {
+    return pattern.regex.test(input)
+  } else {
+    throw new Error(`Invalid pattern at runtime: ${typeof pattern}`)
+  }
+}
+
+export function stringifyPattern (pattern: Pattern) {
+  if (typeof pattern === 'string') {
+    return pattern;
+  } else if ('regex' in pattern) {
+    return `{regex: ${pattern.regex}}`
+  } else {
+    return '[invalid pattern]'
+  }
+}
+
+export const regexDecoder = string().map(value => new RegExp(value))
```

Comment:
Oh, nevermind my first comments. You actually aren't making the regex comparisons case insensitive.

This is weird cuz I think in general people may be more likely to want a case insensitive check than a case sensitive one.
On the other hand, it would be an implicit behavior that users may not want, and that they could implement on their own via the regex pattern itself.
However, the config file maybe could also take regex options? 😁 Not sure how we'd write them, but something like `- regex: 'my_regex', options: gixmUuJD`. Or maybe like in other languages, the pattern could be written such as `- regex: '/my_regex/gixmUuJD'`? 🤔 not sure if this is supported natively in typescript.

Reply:
Initially I'm not interested in supporting regex options. It's fine to have case-sensitive matching for labels for now. Options (and discussions about options) can be added in a separate PR.

Reply:
SGTM